addons/pkg-maven/common/src/main/java/org/commonjava/indy/pkg/maven/content/MetadataUtil.java

```diff
@@ -0,0 +1,30 @@
+package org.commonjava.indy.pkg.maven.content;
+
+import org.commonjava.indy.model.core.StoreKey;
+import org.commonjava.indy.subsys.infinispan.CacheHandle;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class MetadataUtil
+{
+    public static void removeAll( StoreKey storeKey, CacheHandle<MetadataCacheKey, MetadataInfo> cacheHandle )
+    {
+        cacheHandle.executeCache( ( cache ) -> {
```

Comment:
We might need to use a query for this if the metadata cache gets too large. I'm not sure whether ISPN can load keys separately from values or not, but that would be a lot of entry data to iterate through (and large database overhead) if we have to do both. This has bitten us before with NFC and content-index, I believe...but it's also worth noting that those caches are MUCH bigger in terms of entry count.

Reply:
This comment is inspiring. Actually, we always stumble and trip when it comes to querying ISPN. I guess ISPN cannot load keys without values. I think it will load the whole entry when we iterate it. What is more, ISPN can not query by field in the key object. It can only query by value objects. One workaround is to add those fields into the value object. This kind of overlap can save us in some cases. But ... if we persist it into db, the query is even harder. We use a string-keyed table to persist the cache where the values are in byte blocks. Once we want to query it, it has to deserialize and load all of them from db (otherwise I cannot imagine how it queries). 
There is still a way out. I am thinking to use a dual-cache. One cache for the key and one for the value, and use checksum string to relate them. E.g, cache1<md5str, keyObject> and cache2<md5str,valueObject>. We can query by the cache1 which is smaller. And use the md5str to get/delete from the second cache. WDYT?